app/assets/javascripts/discourse/app/components/instant-search-container.gjs

```diff
@@ -0,0 +1,28 @@
+import Component from "@glimmer/component";
+import AisInstantSearch from "@discourse/ember-instantsearch";
```

Comment:
If we wanna make all of these available together in a single async import, we need a 'barrel file'. Perhaps it should be an `index.js` file in the `components/` directory of ember-instantsearch? Then you could do something like

```js
const { AisClearRefinements, AisCurrentRefinements } = await import("@discourse/ember-instantsearch/components");
```

`components/index.js` would be something like

```js
export { default as AisClearRefinements } from "./ais-clear-refinements";
export { default as AisCurrentRefinements } from "./ais-current-refinements";
```


> service

Ahhh... that complicates things, because Ember doesn't currently provide any way to lazy-load services like this 😬  (it's hopefully coming in Polaris). Looking at the service though... it doesn't seem to use any application state, so perhaps it doesn't need to be a service at all. Could it just be a simple `lib` module, exporting `snippet()` and `highlight()` functions?

All that said... this is quite complex juggling for an experiment, especially since it doesn't sound like we'll be using instantsearch long term. How about just copy/pasting the components/services into the Discourse plugin you're working on? They won't be async loaded. But at least they will only be loaded when the experimental plugin is installed/enabled.

That would also make iterating on the implementation much quicker, since re-releasing the npm module (and then bumping in core) won't be required.